docs/tutorials/simple_trading.py.md

```diff
@@ -0,0 +1,142 @@
+class SimpleMarketListener(MarketStreamListener):
+  def on_instrument_data(self, instrument_data):
+    global selected_futures_id
+    futures = [instrument for instrument in instrument_data['data'].values() if instrument['type'] == 'futures'][0]
+    selected_futures_id = futures['instrument_id']
+
+  def on_order_book(self, order_book):
+    if order_book['instrument_id'] != selected_futures_id:
+      return 
+    bids = order_book['bids']
+    if bids and (not bids[0] or float(bids[0][0]) > sell_threshold):
+      user_stream.place_order({
+        'instrument_id': selected_futures_id, 
+        'client_order_id':  int(time() * 1000000),
+        'side': 'sell',
+        'quantity': 1000,
+        'limit_price': bids[0][0],
+        'order_type': 'limit',
+      })
+```
+
+We've implemented only two methods of `MarketStreamListener`, but we could implement more, if we wanted to make decisions
+based on different data (e.g. `on_quotes`, `on_trade`, cf. `MarketStreamListener` documentation).
+
+## 3. Risk Control
+
+Tight risk control is essential in every algo strategy. We will now define `UserStreamListener` which will record
+every open position (cf. `on_open_position`) and throw them on the market if balance of our account falls bellow
+3.1415927 BTC (cf. `on_account_state`).
+
+```python
+//== user_stream_listener ==//
+
+open_positions = {}
+balance_threshold = 3.1415927
+
+class SimpleUserListener(UserStreamListener):
+  def on_open_position(self, open_position):
+    open_positions[open_position['instrument_id']] = open_position
+    
+  def on_account_state(self, account_state):
+    if account_state['balance'] < balance_threshold:
+       # panic
+       orders = []
+       for open_position in open_positions.values():
+        order_side = 'buy' if open_position['side'] == 'short' else 'sell'
+        orders.append({
+          'type': 'place_order',
+          'instrument_id': open_position['instrument_ud'], 
+          'client_order_id':  int(time() * 1000000),
+          'side': order_side,
+          'quantity': open_position['quantity'],
+          'limit_price': '0.00000001' if order_side == 'sell' else '100000',
+          'order_type': 'limit',
+        })
+       self.user_stream.batch(orders)
+```
+
+Again, we've implemented only two methods of `UserStreamListener`, but in a real-life scenario we would also have to
+control which of our orders get placed (cf. `on_order_placed`/`on_order_place_failed`), which get filled (cf. 
+`on_order_filled`), etc.
+
+## 4. Exchange, Trader
+
+```python
+quedex_public_key = open("quedex-public-key.asc", "r").read()
+exchange = Exchange(quedex_public_key, 'wss://api.quedex.net')
+
+trader_private_key = open("trader-private-key.asc", "r").read()
+account_id = open("account-id", "r").read()
+trader = Trader(trader_private_key, '83745263748')
+trader.decrypt_private_key('s3cret')
+```
+
+## 5. Setting up streams
+
+```python
+
+user_stream = UserStream(exchange, trader, SimpleUserListener)
```

Comment:
I vote for option 1, but:
a) It'd be good to have `.addListener` and `.removeListener` methods, so that algotrading functionalities may be enabled/disabled by adding/removing listeners,
b) I'd like to have fully reflection-based API in which no inheritance is required.